components/clp-package-utils/clp_package_utils/scripts/stop_clp.py

```diff
@@ -109,47 +113,47 @@ def main(argv):
         with open(instance_id_file_path, "r") as f:
             instance_id = f.readline()
 
-        if "" == component_name or WEBUI_COMPONENT_NAME == component_name:
+        if target in (ALL_TARGET, CONTROLLER_TARGET_NAME, WEBUI_COMPONENT_NAME):
             stop_container(f"clp-{WEBUI_COMPONENT_NAME}-{instance_id}")
-        if "" == component_name or SEARCH_WORKER_COMPONENT_NAME == component_name:
+        if target in (ALL_TARGET, CONTROLLER_TARGET_NAME, SEARCH_WORKER_COMPONENT_NAME):
```

Comment:
Is the intention that you don't want users to stop the controller components without stopping the other components? If so, I'd say that's not something to address in this PR since:
* It's trying to add a level of intelligence that the rest of the script doesn't respect (e.g., you can still stop an individual controller component without stopping a worker).
* It's not strictly necessary. For instance, it's a valid operation to stop the controller components but not stop the database.